### PR TITLE
Convert between (Int|UInt)(16|32|64) and Data while respecting endianness

### PR DESCRIPTION
**Pull request by cprime**

### CODE REVIEW

File: SwiftWisdomTests/Data/DataConversionTests.swift [hunk 1]
```diff
@@ -50,7 +50,25 @@ class DataConversionTests: XCTestCase {
         let data = Data(byte: 42)
         XCTAssert(data.ip_uint8Value == 42)
     }
-    
+
+    func testUInt16() {
+        let data = Data(bytes: [0x00, 0x01])
+        XCTAssert(data.ip_bigEndianUInt16Value == 1)
+        XCTAssert(data.ip_littleEndianUInt16Value == UInt16(1 << 8))
```
Comment (paulrolfe) [inline SwiftWisdomTests/Data/DataConversionTests.swift]: Can we also test when a nil should be returned in these?

**Review by paulrolfe — CHANGES_REQUESTED** [review]
I like it. Not necessary for this PR, but maybe it should  also exist on signed Int16, 32, 64?

**Comment by cprime** [comment]
@paulrolfe I've added Int16, 32, 64 support as well

**Review by paulrolfe — APPROVED** [review]

**Comment by paulrolfe** [comment]
retest this please

**Comment by cprime** [comment]
retest this please

**Comment by paulrolfe** [comment]
Looks like some tests need to be fixed.

File: SwiftWisdom/Core/StandardLibrary/Numbers/Integer+Extensions.swift [hunk 2]
```diff
@@ -56,3 +56,32 @@ extension NumberFormatter {
         return decimalFormatter
     }
 }
+
+extension Integer {
+    fileprivate var ip_data: Data {
```
Comment (brightredchilli) [inline SwiftWisdom/Core/StandardLibrary/Numbers/Integer+Extensions.swift:61]: I would rather see this as a Data constructor.

**Review by brightredchilli — APPROVED** [review]

**Comment by brightredchilli** [comment]
what was the change, ps?

File: SwiftWisdom/Core/Foundation/Data/Data+Extensions.swift [hunk 3]
```diff
@@ -20,7 +20,8 @@ extension Data {
     // From here: http://stackoverflow.com/questions/38023838/round-trip-swift-number-types-to-from-data
     public func to<T>(type: T.Type) -> T? {
         guard self.count > 0 else { return nil }
-        return self.withUnsafeBytes { $0.pointee }
```
Comment (cprime) [inline SwiftWisdom/Core/Foundation/Data/Data+Extensions.swift:23]: These changes fixed the non-deterministic test failures

File: SwiftWisdom/Core/StandardLibrary/Numbers/Integer+Extensions.swift (hunk 2 above)
Comment (cprime) [reply SwiftWisdom/Core/StandardLibrary/Numbers/Integer+Extensions.swift:61]: This is more inline with the `var ip_data: Data` on all of the other primitive types

File: SwiftWisdom/Core/Foundation/Data/Data+Extensions.swift (hunk 3 above)
Comment (cprime) [reply SwiftWisdom/Core/Foundation/Data/Data+Extensions.swift:23]: @brightredchilli

Comment (cprime) [reply SwiftWisdom/Core/Foundation/Data/Data+Extensions.swift:23]: no idea why

**Comment by cprime** [comment]
retest this please

**Comment by cprime** [comment]
retest this please
